src/twisted/words/protocols/irc.py

```diff
@@ -88,7 +88,7 @@ def parsemsg(s):
     trailing = []
     if not s:
         raise IRCBadMessage("Empty line.")
-    if s[0] == ':':
+    if s[0:1] == ':':
```

Comment:
It's not 100% necessary, but it still works.
I was playing around with byte strings in different parts of the code earlier,
and found that:

```
foo = b"blah"
foo[0]   # Returns b'b' on Python 2, and 102 on Python 3 
foo[0:1] # Returns b'b' on Python 2 and Python 3
```

For native strings:

```
bar = "bar"
bar[0]  #returns "b" on Python 2 and Python 3
bar[0:1] # returns "b" on Python 2 and Python 3
```